packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -50,6 +50,19 @@ export abstract class AbstractResourcePreferenceProvider implements PreferencePr
         return this.preferences;
     }
 
+    async setPreference(key: string, value: string): Promise<void> {
+        const resource = await this.resource;
+        if (resource.saveContents) {
+            const content = await resource.readContents();
+            const formattingOptions = {tabSize: 3, insertSpaces: true, eol: ''};
```

Comment:
What about detecting it from the original content with `detect-indent`? It is not nice to override user or workspace settings.

Reply:
`jsoncparser` takes care about it, this options is needed if indent was not detected e.g. empty file